src/nbl/asset/utils/ISPIRVDebloater.cpp

```diff
@@ -0,0 +1,115 @@
+#include "nbl/asset/utils/ISPIRVDebloater.h"
+#include "nbl/asset/utils/ISPIRVOptimizer.h"
+#include "nbl_spirv_cross/spirv.hpp"
+
+#include "nbl/core/declarations.h"
+#include "nbl/system/ILogger.h"
+
+
+using namespace nbl::asset;
+
+ISPIRVDebloater::ISPIRVDebloater()
+{
+    constexpr auto optimizationPasses = std::array{
+        ISPIRVOptimizer::EOP_ELIM_DEAD_FUNCTIONS,
+        ISPIRVOptimizer::EOP_ELIM_DEAD_VARIABLES,
```

Comment:
you may need EOP_REMOVE_UNUSED_INTERFACE_VARIABLE_PASS and others because dead variables may only remove globals, descriptors, push constants and groupshared (read the source/docs)